app/stac_api/views_mixins.py

```diff
@@ -26,21 +32,32 @@ def get_write_request_data(self, request, *args, **kwargs):
 
     def create(self, request, *args, **kwargs):
         data = self.get_write_request_data(request, *args, **kwargs)
+        if is_asset(type(self)):
+            asset_link_self = request.build_absolute_uri() + "/" + request.data["id"]
+        else:
+            asset_link_self = None
         serializer = self.get_serializer(data=data)
         serializer.is_valid(raise_exception=True)
         self.perform_create(serializer)
-        headers = self.get_success_headers(serializer.data)
+        headers = self.get_success_headers(serializer.data, asset_link_self)
         return Response(serializer.data, status=status.HTTP_201_CREATED, headers=headers)
 
     def perform_create(self, serializer):
         serializer.save()
 
-    def get_success_headers(self, data):
-        try:
-            return {'Location': get_link(data['links'], 'self', raise_exception=True)['href']}
-        except KeyError as err:
-            logger.error('Failed to set the Location header for item creation: %s', err)
-            return {}
+    def get_success_headers(self, data, asset_link_self=None):
+        if not is_asset(type(self)):
+            try:
+                return {'Location': get_link(data['links'], 'self', raise_exception=True)['href']}
+            except KeyError as err:
+                logger.error('Failed to set the Location header for item creation: %s', err)
+                return {}
+        else:
+            try:
+                return {'Location': asset_link_self}
+            except KeyError as err:
+                logger.error('Failed to set the Location header for asset creation: %s', err)
```

Comment:
The handling of location could be easily done within `AssetsList` class  (and should be) by overriding `get_success_headers()` method. This method in `CreateModelMixin` should simply be extended with the `request` parameter.

But a fundamental questions we may be better add `links` to the Asset like in Collection and Item, this way our API is more consistent, what do you think @boecklic and @rebert  ? The asset is anyway swisstopo specific and not part of the original STAC we can define it as we want.

Reply:
Good point, I'll implement the handling of the location in the `get_success_headers()` inside `AssetList`.
As for the general question of including links to the assets like in collection and item, I'd suggest we discuss this and - if agreed - create a separate ticket for this (in order to get this PR merged soon).
From my point of view we could add links to be consistent. It shouldn't be a big deal and probably is helpful for users of the browseable API as well.